plugin/kotel/metrics.go

```diff
@@ -0,0 +1,217 @@
+package kotel
+
+import (
+	"context"
+	"go.opentelemetry.io/otel/attribute"
+	semconv "go.opentelemetry.io/otel/semconv/v1.12.0"
+	"math"
+	"net"
+	"strconv"
+	"time"
+
+	"github.com/twmb/franz-go/pkg/kgo"
+	"go.opentelemetry.io/otel/metric"
+	"go.opentelemetry.io/otel/metric/global"
+	"go.opentelemetry.io/otel/metric/instrument"
+	"go.opentelemetry.io/otel/metric/instrument/syncint64"
+	"go.opentelemetry.io/otel/metric/unit"
+)
+
+var ( // interface checks to ensure we implement the hooks properly
+	_ kgo.HookBrokerConnect       = new(Meter)
+	_ kgo.HookBrokerDisconnect    = new(Meter)
+	_ kgo.HookBrokerWrite         = new(Meter)
+	_ kgo.HookBrokerRead          = new(Meter)
+	_ kgo.HookProduceBatchWritten = new(Meter)
+	_ kgo.HookFetchBatchRead      = new(Meter)
+)
+
+type Meter struct {
+	provider metric.MeterProvider
+	meter    metric.Meter
+	metrics  Metrics
+}
+
+// MetricsOption interface used for setting optional config properties.
+type MetricsOption interface {
```

Comment:
To be a bit more consistent with other areas of franz-go, can `Option` be renamed `Opt`?

Also, since this is configuring a `Meter`, should this be `MeterOpt`?